src/main/java/net/pms/dlna/DLNAResource.java

```diff
@@ -3958,53 +3726,438 @@ protected void setDiscovered(boolean discovered) {
+				return;
+			}
+
+			List<DLNAMediaSubtitle> subtitlesList = media.getSubtitleTracksList();
+			hasSubtitles = !subtitlesList.isEmpty();
+			for (DLNAMediaSubtitle subtitles : subtitlesList) {
+				if (subtitles.isExternal()) {
+					hasExternalSubtitles = true;
+					break;
+				}
+			}
+			isExternalSubtitlesParsed = true;
+		}
+	}
+
+	/**
+	 * This method figures out which audio track should be used based on
+	 * {@link DLNAMediaInfo} metadata and configuration settings.
+	 *
+	 * @param renderer the {@link RendererConfiguration} from which to get the
+	 *            configuration or {@code null} to use the default
+	 *            configuration.
+	 * @return The resolved {@link DLNAMediaAudio} or {@code null}.
+	 */
+	public DLNAMediaAudio resolveAudioStream(RendererConfiguration renderer) {
+		if (media == null || media.getAudioTrackCount() == 0) {
+			LOGGER.trace("Found no audio track");
+			return null;
+		}
+		// Use device-specific pms conf
+		PmsConfiguration deviceSpecificConfiguration = PMS.getConfiguration(renderer);
+
+		// check for preferred audio
+		DLNAMediaAudio dtsTrack = null;
+		StringTokenizer st = new StringTokenizer(deviceSpecificConfiguration.getAudioLanguages(), ",");
+		while (st.hasMoreTokens()) {
+			String lang = st.nextToken().trim();
+			LOGGER.trace("Looking for an audio track with language \"{}\"", lang);
+			for (DLNAMediaAudio audio : media.getAudioTracksList()) {
+				if (audio.matchCode(lang)) {
+					LOGGER.trace("Matched audio track: {}", audio);
+					return audio;
+				}
+
+				if (dtsTrack == null && audio.isDTS()) {
+					dtsTrack = audio;
+				}
+			}
+		}
+
+		// preferred audio not found, take a default audio track, dts first if available
+		if (dtsTrack != null) {
+			LOGGER.trace("Preferring DTS audio track since no language match was found: {}", dtsTrack);
+			return dtsTrack;
+		}
+		DLNAMediaAudio result = media.getFirstAudioTrack();
+		LOGGER.trace("Using the first available audio track: {}", result);
+		return result;
+	}
+
+	/**
+	 * This method figures out which subtitles track should be used based on
+	 * {@link DLNAMediaInfo} metadata, chosen audio language and configuration
+	 * settings.
+	 *
+	 * @param renderer the {@link RendererConfiguration} from which to get the
+	 *            configuration or {@code null} to use the default
+	 *            configuration.
+	 * @param audioLanguage the {@code ISO 639} language code for the chosen
+	 *            audio language or {@code null} if it doesn't apply.
+	 * @param forceRefresh if {@code true} forces a new scan for external
+	 *            subtitles instead of relying on cached information (if it
+	 *            exists).
+	 * @return The resolved {@link DLNAMediaSubtitle} or {@code null}.
+	 */
+	public DLNAMediaSubtitle resolveSubtitlesStream(RendererConfiguration renderer, String audioLanguage, boolean forceRefresh) {
+		if (media == null) {
+			return null;
+		}
+
+		// Use device-specific pms conf
```

Comment:
"pms configuration" or "UMS.conf"

Reply:
@Sami32 Again not my text. I think it refers to the class "PmsConfiguration" in a lazy way.